File: backend-nest/src/seed.ts
import { NestFactory } from '@nestjs/core';
import { getModelToken } from '@nestjs/mongoose';
import { Model } from 'mongoose';

import { AppModule } from './app.module';
import { hashPassword } from './auth/password';
import { User } from './users/user.schema';

const PASSWORD = 'password123';

const USERS = [
  ['Aarti Rao', 'aarti@example.com'],
  ['Dev Menon', 'dev@example.com'],
  ['Kiran Shah', 'kiran@example.com'],
  ['Meera Iyer', 'meera@example.com'],
  ['Rohit Nair', 'rohit@example.com'],
];

async function seed() {
  const app = await NestFactory.createApplicationContext(AppModule, {
    logger: ['error', 'warn'],
  });
  const userModel = app.get<Model<User>>(getModelToken(User.name));
  const now = new Date();

  await userModel.init();
  await userModel.collection.drop();
  await userModel.syncIndexes();
  await userModel.insertMany(
    await Promise.all(
      USERS.map(async ([name, email]) => ({
        name,
        email,
        password_hash: await hashPassword(PASSWORD),
        initials: name
          .split(' ')
          .map((part) => part[0])
          .join(''),
        created_at: now,
      })),
    ),
  );

  await app.close();
  console.log(
    `Seeded ${USERS.length} users into '${userModel.db.name}'. Password for all: ${PASSWORD}`,
  );
}

void seed();

File: backend/app/seed.py
import asyncio
from datetime import UTC, datetime

from app.auth.security import hash_password
from app.db import client, db

PASSWORD = "password123"

USERS = [
    ("Aarti Rao", "aarti@example.com"),
    ("Dev Menon", "dev@example.com"),
    ("Kiran Shah", "kiran@example.com"),
    ("Meera Iyer", "meera@example.com"),
    ("Rohit Nair", "rohit@example.com"),
]


async def seed() -> None:
    now = datetime.now(UTC)
    await db.users.drop()
    await db.users.create_index("email", unique=True)
    await db.users.insert_many(
        {
            "name": name,
            "email": email,
            "password_hash": hash_password(PASSWORD),
            "initials": "".join(part[0] for part in name.split()),
            "created_at": now,
        }
        for name, email in USERS
    )
    await client.close()
    print(f"Seeded {len(USERS)} users into '{db.name}'. Password for all: {PASSWORD}")


if __name__ == "__main__":
    asyncio.run(seed())
